app/controllers/api/v1/orders_controller.rb

```diff
@@ -14,6 +14,20 @@ def show
                  status: 404
         end
       end
+
+      def index
+        order_references = Order::Reference.where(search_query)
+
+        render :index, locals: { order_references: order_references }
+      end
+
+      private
+      
+      def search_query
+        OrderQuery.new(warehouse_id: params[:warehouse_id],
```

Comment:
Did you not like the idea of using ransack for this, like I have with the contract model and controller?